### ISSUE TITLE
AOT compatibility

### ISSUE THREAD

**Issue by cacowen** [stripe/stripe-dotnet#2967]
### Is your feature request related to a problem? Please describe.

Currently the workflow to capture Webhooks and other models depends on direct JSON transformations and furthermore relies on Newtonsoft. These add a lot of overhead in my otherwise simple Lambda function. And pretty much eliminates the AOT benefits.

### Describe the solution you'd like

I would like to be able to use `stripe-dotnet` in a .Net 8 AOT AWS Lambda Function without bypassing trimming and other AOT features. It would be nice to have stripe models that do not rely on Newtonsoft and can easily be added to a JSON serializer context.

### Additional context

Perhaps I am just using this the wrong way. I have tried to follow examples here and in YouTube (including the Stripe Developers channel). So far, I have not been able to use this in an AOT application in a straightforward and clean way.

**Comment by jar-stripe** [comment]
Hi @cacowen, thanks for the request!  I can definitely see the use case here.  We're planning to look into using System.Text.Json as an alternative to Newtonsoft soon, and I suspect we'll have to get thru at least that work before we can try to understand what we need to do to support this.  Unfortunately I can't make commitments about prioritization in the future, but we are tracking it.  Thanks again!

**Comment by charlesroddie** [comment]
As of https://github.com/stripe/stripe-dotnet/pull/3327 System.Text.Json has been added. Presumably the next step after that will be to remove NewtonSoft.Json. And then you can see what trim warnings remain? Is that the right plan @jar-stripe ?

**Comment by jar-stripe** [comment]
Hi @charlesroddie thanks for the note!  That is roughly the plan but we are still evaluating our options to avoid breaking users who still use Newtonsoft.  We will update this ticket when we have more info here!

**Comment by charlesroddie** [comment]
@jar-stripe would you accept a PR for this?

If not we will have to use a fork or do things without the dotnet wrapper.

**Comment by jar-stripe** [comment]
@charlesroddie Thanks for the offer and appreciate the context!  Unfortunately, most of the code here is generated so we cannot accept a PR.  I have not had a chance to look at how .NET AOT works; can you describe what change you would make?  E.g. is it just removing Newtonsoft as a dependency and removing the corresponding annotations?  Or does AOT require other scaffolding to work?

**Comment by charlesroddie** [comment]
I did a spike to investigate this. There are two parts.

A. Dropping Newtonsoft. This needs to be done in your code generator, removing 18000 JsonProperty/JsonConverter properties. Removing the attributes is not a breaking change, since callers never reference them. Deleting the public members typed with Newtonsoft types is technically breaking, but you have already obsoleted most of them and the remainder are similarly obsoletable.

B. Doable in this repo:
1. replacing the reflection-based serialier in Infrastructure/ with a Roslyn source generator that reads the STJ attributes already on the models
2. taking Newtonsoft off the live request path (two headers and V2 thin events)
3. deleting unused discriminated union converters and FormEncoder

A and B are independent.

B has a size impact (increasing the dll from 5MB to 18MB). Most of that difference is likely trimmable, but some is not unless the StripeRypeRegistry.ObjectToTypes and GetConcreteType methods can be reworked.
